src/util/log.c

```diff
@@ -0,0 +1,648 @@
+                                 MSG_NOSIGNAL);
+                        if (l < 0)
+                                return error_origin(-errno);
+                }
+
+                return 0;
+        } else if (errno != EMSGSIZE) {
+                return error_origin(-errno);
+        }
+
+        /*
+         * We could not send the message as a single datagram. Instead, we now
+         * seal the memfd and send it as payload of an empty datagram. The
+         * journal can deal with this and treats the memfd content as log
+         * message.
+         *
+         * Note that this means there is an inflight memfd pending on the
+         * journal that is accounted on us as long as the journal did not read
+         * it, yet. Hence, clients should never be able to trigger such log
+         * messages, otherwise they could exploit this and make us exceed our
+         * inflight FD limit.
+         *
+         * We could send a pipe-fd as barrier and block on it. Hence, we would
+         * be able to know at which point our messages are no longer inflight.
+         * However, this would be a quite expensive code-path, making this
+         * entire path completely useless.
+         *
+         * Long story short: Do not write excessive log messages, unless
+         *                   debugging is enabled in some way.
+         */
+
+        mfd = log->mem_fd;
+        log->mem_fd = -1;
+        munmap(log->map, LOG_SIZE_MAX);
+        log->map = MAP_FAILED;
+
+        r = ftruncate(mfd, log->offset);
+        if (r < 0)
+                return error_origin(-errno);
+
+        /*
+         * XXX: We hard-code F_ADD_SEALS and the seal-bitmask here, since we
+         *      cannot include <linux/fcntl.h> when we already
+         *      include <fcntl.h>... Give it up for shared namespaces!
+         *
+         *          fcntl(mfd,
+         *                F_ADD_SEALS,
+         *                F_SEAL_SHRINK | F_SEAL_GROW
+         *                              | F_SEAL_WRITE
+         *                              | F_SEAL_SEAL);
+         */
+        r = fcntl(mfd, 1033, 0xf);
+        if (r < 0)
+                return error_origin(-errno);
+
+        r = log_fd_send(log->log_fd, mfd);
+        if (r < 0)
+                return error_trace(r);
+
+        return 0;
+}
+
+static int log_commit_stderr(Log *log, const char *format, va_list args) {
+        int r;
+
+        /*
+         * Lets format the log-message in our staging buffer and print it out.
+         * Note that we don't support structured logging. Hence, we simply
+         * ignore anything that is in the staging buffer and instead just print
+         * our message.
+         * Sadly, `%r' is still no standardized format specifier, so we need
+         * the separate calls into log_append() here.
+         */
+
+        log->offset = 0;
+
+        if (format && *format) {
+                log_appendf(log, "<%d>", log->level ?: 6);
```

Comment:
Why the magic number? Make a define? Is this not LOG_INFO? (same elsewhere). This is basically the same logic as in log_append_common(), right?